src/pages/index.js

```diff
@@ -24,24 +24,191 @@ const profileDescriptionInput = document.querySelector(
+const addCardFormValidator = new FormValidator(settings, addCardForm);
+const avatarFormValidator = new FormValidator(settings, avatarForm);
+
+profileEditFormValidator.enableValidation();
+addCardFormValidator.enableValidation();
+avatarFormValidator.enableValidation();
+
+/* Popup Instances */
+const deleteCard = new PopupWithFormSubmit("#trash-popup");
+const imagePopup = new PopupWithImage("#image-popup");
+const avatarPopup = new PopupWithForm("#avatar-edit-popup", handleAvatarSubmit);
+const profilePopup = new PopupWithForm(
+  "#profile-edit-popup",
+  handleProfileEditSubmit
+);
+const cardPopup = new PopupWithForm("#add-card-popup", handleAddCardFormSubmit);
+
+deleteCard.setEventListeners();
+imagePopup.setEventListeners();
+avatarPopup.setEventListeners();
+profilePopup.setEventListeners();
+cardPopup.setEventListeners();
+
+/* Get Initial Cards */
+let cardList;
+api
+  .getInitialCards()
+  .then((initialCards) => {
+    console.log("intialCards", initialCards);
+    cardList = new Section(
+      {
+        items: initialCards,
+        renderer(item) {
+          const cardElement = renderCard(item);
+          this._container.prepend(cardElement);
+        },
+      },
+      ".cards__list"
+    );
+    cardList.renderItems();
+  })
+  .catch((error) => {
+    console.error("Error in getInitialCards:", error);
+  });
+
+api
+  .getUserInfo()
+  .then((userInfo) => {
+    console.log("User information retrieved successfully:", userInfo);
+    userProfile.setUserInfo(userInfo);
+  })
+  .catch((error) => {
+    console.error("Error getting user information:", error);
+  });
+
+/* Profile Image Click Handler */
+const profileImage = document.querySelector(".profile__avatar-edit");
+profileImage.addEventListener("click", () => {
+  avatarFormValidator.resetValidation();
+  avatarPopup.open();
+});
+
+/* Handlers */
 function handleImageClick(cardData) {
   imagePopup.open(cardData);
 }
+
 function handleProfileEditSubmit(inputValues) {
-  userProfile.setUserInfo(inputValues);
+  const updatedData = {
+    name: inputValues.name,
+    about: inputValues.about,
+  };
+
+  // Change button text to "Saving..."
+  profilePopup.setLoadingState(true);
+
+  api
+    .updateUserProfile(updatedData)
+    .then((updatedUserInfo) => {
+      userProfile.setUserInfo(updatedUserInfo);
+      console.log("User profile updated successfully:", updatedUserInfo);
+    })
+    .catch((error) => {
+      console.error("Error updating user profile:", error);
+    })
+    .finally(() => {
+      // Reset button text to its original state
+      profilePopup.setLoadingState(false);
+    });
+}
+function handleAvatarSubmit({ avatar }) {
+  // Change button text to "Saving..."
+  avatarPopup.setLoadingState(true);
+
+  api
+    .updateAvatar({ avatar: avatar })
+    .then((updatedUserInfo) => {
+      console.log("Avatar updated successfully:", updatedUserInfo);
+      document.querySelector(".profile__image").src = updatedUserInfo.avatar;
```

Comment:
`      document.querySelector(".profile__image").src = updatedUserInfo.avatar;
`

Class `UserInfo` should set the entire user’s data (including the **avatar**)

Reply:
* You haven't corrected my remark https://github.com/Domvm-7/se_project_aroundtheus/pull/2/files#r1447791198 

` userProfile.setUserInfo(updatedUserInfo);`